content/source/docs/enterprise/system-overview/architecture.html.md

```diff
@@ -5,24 +5,201 @@ page_title: "Architecture Summary - System Overview - Terraform Enterprise"
+
+1. `ptfe_atlas` detects a state change and enqueues a background job for `ptfe_sidekiq` in Redis.
+2. `ptfe_sidekiq` asynchronously pulls the job and passes the details to `ptfe_build_manager`.
+3. `ptfe_build_manager` enqueues a `ptfe_build_worker` job in RabbitMQ.
+4. `ptfe_build_worker` spins up a temporary environment, runs a Terraform plan or apply, and passes the results to `ptfe_archivist`.
+5. `ptfe_archivist` persists the plan or apply results and returns a URL to `ptfe_build_worker`.
+6. `ptfe_build_worker` passes the plan or apply results back to `ptfe_build_manager`.
+7. `ptfe_build_manager` `POST`s the results to an internal webhook.
+8. `ptfe_nginx` receives the request and forwards the callback to `ptfe_atlas`.
+
+Finally, `ptfe_atlas` receives results of the plan or apply and begins next actions if any apply.
+
+### Sentinel Run
+
+A state machine inside `ptfe_atlas` is responsible for scheduling Sentinel runs when necessary.
+
+![tfe-int-sentinel-run](assets/tfe-int-sentinel-run.png)
+
+1. `ptfe_atlas` detects a state change and queues a background job for `ptfe_sidekiq` in Redis.
+1. `ptfe_sidekiq` asynchronously pulls the job and schedules a Sentinel run in `ptfe_nomad`.
+1. `ptfe_nomad` runs the Sentinel job and `POST`s the result back to `ptfe_atlas`.
+
+Although the result of the Sentinel run is known at this point, it is not persisted until the callback is processed:
+
+![tfe-int-sentinel-persistence](assets/tfe-int-sentinel-persistence.png)
+
+
+1. `ptfe_atlas` receives the callback request from `ptfe_nginx` and streams the response to `ptfe_archivist`.
+2. `ptfe_archivist` saves the Sentinel results to blob storage and returns an Object ID to `ptfe_atlas`.
+
+Finally, `ptfe_atlas` updates the metadata in PostgreSQL after it receives confirmation that the details of the run have been persisted.
+
+## Long-Running Container List
+
+The following table lists every container Terraform Enterprise runs via the Replicated Native Scheduler.  The list is presented by layer as per the above document sections and includes those long-running processes only, not ephemeral containers.
+
+| Layer | Container Name | Process Name | Container Function |
+| - | - | - | - |
+| Web Services | `ptfe_atlas` | Atlas | Ruby on Rails application that powers the UI and API. |
+|  | `ptfe_registry_api` | Private Registry API | Go service that powers the private module registry. |
+|  | `ptfe_nginx` | Nginx | NginX is used as a router in Terraform Enterprise. |
+| Background Services | `ptfe_sidekiq` | Sidekiq | Ruby workers that run asynchronous jobs enqueued by `ptfe_atlas`. |
+|  | `ptfe_state_parser` | Terraform State Parser | Go service that parses Terraform State Files. |
+|  | `ptfe_build_manager` | Terraform Build Manager | Go service that coordinates Terraform runs, plan and applies. |
+|  | `ptfe_registry_worker` | Private Module Registry Worker | Go service that parses Git repositories for Private Registry. |
```

Comment:
I think Terraform Registry is a proper noun, but this isn't. Please correct if if I'm wrong!

```suggestion
|  | `ptfe_registry_worker` | Private Module Registry Worker | Go service that parses Git repositories for private registry. |
```